core/trino-main/src/test/java/io/trino/type/TestIntegerType.java

```diff
@@ -55,6 +56,7 @@ protected Object getGreaterValue(Object value)
     }
 
     @Override
+    @Test
```

Comment:
Can we ensure automatically that this doesn't happen?
The method is silently ignored if not annotated with `@Test`

Reply:
We shouldn’t be overriding tests like that. If necessary, add new tests specific for the type.

Reply:
I don't follow.
This is pre-existing functionality.

Please add more detail to the comment.
Simply create a new method  instead of doing overrides?

Reply:
I'm just responding to your comment about "ensuring automatically it doesn't happen". Structuring tests that way is not a good practice so we should fix them separately and not do that going forward.

Reply:
That is how ATDQ, BCT and other similar "super" test classes are designed to work. We'll need to rethink how to structure those before those can be migrated over safely - since without that it's hard to have re-usable tests which also evolve over time (e.g. `testRange` is overridden today to throw SkipException but in future once the feature is there the override can be removed).

Reply:
cc: @ksobolew any ideas to how to get this to work or how to restructure tests to avoid subclassing when using JUnit5?

Reply:
Well, I can see that the implementations in the base class assert that the relevant properties of the type are `Optional.empty()`. They could, instead, do some assertions if it's not empty, and use some protected methods to get the expected result. But I'm not sure if that would work with all the types. It's kind of similar to how `hasBehavior` is used in connector tests...

Reply:
> That is how ATDQ, BCT and other similar "super" test classes are designed to work. We'll need to rethink how to structure those before those can be migrated over safely

I can think of many different, more "pure" approaches.
The whole idea of BCT is impure and we could remove it. Connectors should have tests for features they implement.

Of course, this is straw-man's proposal. While letting us free of test overrides, this would lead to much worse test coverage.